pkg/checks/podRestarts/podRestarts_test.go

```diff
@@ -1,18 +1,118 @@
+	c.RestartObservations["ssPod"] = append(c.RestartObservations["ssPod"], RestartCountObservation{time.Now().Add(time.Duration(-2)*time.Minute), 26})
+	c.RestartObservations["ssPod"] = append(c.RestartObservations["ssPod"], RestartCountObservation{time.Now().Add(time.Duration(-1)*time.Minute), 27})
+	c.RestartObservations["ssPod"] = append(c.RestartObservations["ssPod"], RestartCountObservation{time.Now(), 0})
+
+	//Simulate a deleted pod
+	c.RestartObservations["deletedPod"] = append(c.RestartObservations["deletedPod"], RestartCountObservation{time.Now(), 0})
+
+	//Simulate an old observation on another normal pod
+	c.RestartObservations["anotherPod"] = append(c.RestartObservations["anotherPod"], RestartCountObservation{time.Now().Add(time.Duration(-70)*time.Minute), 15})
+	c.RestartObservations["anotherPod"] = append(c.RestartObservations["anotherPod"], RestartCountObservation{time.Now().Add(time.Duration(-62)*time.Minute), 16})
+	c.RestartObservations["anotherPod"] = append(c.RestartObservations["anotherPod"], RestartCountObservation{time.Now().Add(time.Duration(-61)*time.Minute), 16})
+	//Simulate a normal observation on a another normal pod
+	c.RestartObservations["anotherPod"] = append(c.RestartObservations["anotherPod"], RestartCountObservation{time.Now().Add(time.Duration(-3)*time.Minute), 17})
+	c.RestartObservations["anotherPod"] = append(c.RestartObservations["anotherPod"], RestartCountObservation{time.Now().Add(time.Duration(-2)*time.Minute), 18})
+	c.RestartObservations["anotherPod"] = append(c.RestartObservations["anotherPod"], RestartCountObservation{time.Now().Add(time.Duration(-1)*time.Minute), 19})
+	c.RestartObservations["anotherPod"] = append(c.RestartObservations["anotherPod"], RestartCountObservation{time.Now(), 20})
+
+
+	//Simulate a list of existing pods returned from a pod list
+	l := &v1.PodList{
+		Items: []v1.Pod{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "normalPod",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "ssPod",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "anotherPod",
+				},
+			},
+		},
+	}
+
+	//Pods that existed before the reap
+	t.Log("Before")
+	for podName, restartObservations := range c.RestartObservations {
+		t.Log(podName)
+		for _, observation := range restartObservations{
+			t.Log(observation)
+		}
+	}
+
+	//Dont fear the reaper
+	c.reapPodRestartChecks(l)
+
+	//Pods that existed after the reap
+	//TODO Replace with an actual programatic check vs this eyeball test
+	//TODO check that each pod group has the expected values after reaping and that the mostRecent and second most Recent
+	//TODO values are correctly filled out
+	t.Log("After")
+	for podName, restartObservations := range c.RestartObservations {
+		t.Log(podName)
+		for _, observation := range restartObservations{
+			t.Log(observation)
+		}
+	}
+}
```

Comment:
the test never fails with `t.Fail` and thus won't actually be useful in automated testing.  Throw in a couple `if x != y { t.FailNow("test failed")}` style entries to make this really nice

Reply:
I had a TODO just a few lines above for exactly that.  I'll just put it in this PR though.